### PR TITLE
YAML: Fix printing doubles a blank line before a comment

### LINKED ISSUE #9130: Unexpectedly doubles a blank line before a YAML comment (2.1.0 regression)
**Prettier 2.1.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBaABAMwhJ6AMAOlOugEYCGATngIzHGkDE5FAXngEzEYCOArvwAeeAMzEQAGhAQADjACW0AM7JQ1KhADuABWoJVKCgBstFAJ6rpZKhTABrODADKsuwqgBzZDCr840nAAtmRwACZh4QAyFF78FJ5wAGIQVEEUMIpeyCAU-DAQUiAAFjBBxgDqxQrwym5gcM4GNQoAbjXmOWDKViAeynBUMDq2nunImCYD0gBWykIAQrYOTs4UQXBRHnATUwEgc0LOHp7GcACK-BDwu8bTIG5UA1Q55uvGRbJUHjAVCmEwYrIAAc+GkXwgAwqtlkOS+cGerR20gE1zgIzkhlyylQUDg4XCRSocAECmJIwS4yQkzu+wGQQUPj8dJOZ0uaNu9xgFDIfwBQKQnGkvgoCmMJwAwhAglSQAiAKxFfgDAAqPMMNPurX8AEkoJFYM4wN95ABBfXOGDmM6cuAAXztQA)
```sh
--parser yaml
```

**Input:**
```yaml
- foo: 0
  bar: 1

  # baz: 2
- quux: 3

```

**Output:**
```yaml
- foo: 0
  bar: 1


  # baz: 2
- quux: 3

```

**Expected behavior:**

Prettier should preserve the single blank line, not add a second blank line.

I bisected this issue to commit ea84392af27720b4b3f99556042d1abf5bd8d419 YAML: Preserve blank lines in comments (#8356). Cc @fisker.

### PR DESCRIPTION
**Pull request by sosukesuzuki**
Fixes #9130

Printing YAML comments be seem still unstable. I will create Issues after this PR is merged. But I think this is enough to solve the regression.

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
**Prettier pr-9143**
[Playground link](https://deploy-preview-9143--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBaABOjBDAOlfTdAYi3QCN9DNSMxr0GM4qoMy8CojaLWey9LiAA0ICAAcYAS2gBnZKGwAnZRADuABRUIFKbABt12AJ4Kx5ZdjABrODADKE69KgBzZDGUBXOGLgAtuRwACYhoQAy2O7e2G5wAGIQygHYMDLuyCDY3jAQoiAAFjABBgDqhdLwcs5gcA66VdIAblUmWWBy5iCucnDKMJpWbqnIAGaGfWIAVnIAHgBCVrb2DtgBcBGucOOTfiCzcw6ubgZwAIreEPC7BlMgzsp9ylkm6wYFEsquMGXSITBCsgABwABjEXwgfTKVgkWS+cGezR2YgAjld4ENJHpsnJUFA4KFQgVlHB0dJSUM4qMkBM7vs+gFpJ4fAyTmdLtcdrS9mIYNhyH8AUCkAAmPlWaQGE4AYQgARpIERAFYCt4+gAVAV6On3Zq+ACSUHCsAcYG+UgAgsaHDATGdbn0AL5OoA)
```sh
--parser yaml
```

**Input:**
```yaml
-  - a

   # - b

   # - c
   
   - e

-  - a

   # - b

   # - c


```

**Output:**
```yaml
- - a

  # - b

  # - c

  - e

- - a
  # - b             <--  Missing line break before

  # - c


```

**Comment by sosukesuzuki** [comment]
@fisker This is a not regression because happens on 2.0.5:

スクリーンショット 2020-09-04 15 53 42

(Sorry for console written in Japanese)

**Comment by fisker** [comment]
No, but I mean to fix both of them in #8356

**Prettier pr-8356**
[Playground link](https://deploy-preview-8356--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBaABOjBDAOlfTdAYi3QCN9DNSMxr0GM4qoMy8CojaLWey9KCAA0ICAAcYAS2gBnZKGwAnZRADuABRUIFKbABt12AJ4Kx5ZdjABrODADKE69KgBzZDGUBXOGLgAtuRwACYhoQAy2O7e2G5wAGIQygHYMDLuyCDY3jAQoiAAFjABBgDqhdLwcs5gcA66VdIAblUmWWBy5iCucnDKMJpWbqnIAGaGfWIAVnIAHgBCVrb2DtgBcBGucOOTfiCzcw6ubgZwAIreEPC7BlMgzsp9ylkm6wYFEsquMGXSITBCsgABwABjEXwgfTKVgkWS+cGezR2YgAjld4ENJHpsnJUFA4KFQgVlHB0dJSUM4qMkBM7vs+gFpJ4fAyTmdLtcdrS9mIYNhyH8AUCkAAmPlWaQGE4AYQgARpIERAFYCt4+gAVAV6On3Zq+ACSUHCsAcYG+UgAgsaHDATGdbn0AL5OoA)
```sh
--parser yaml
```

**Input:**
```yaml
-  - a

   # - b

   # - c
   
   - e

-  - a

   # - b

   # - c

```

**Output:**
```yaml
- - a

  # - b

  # - c

  - e

- - a

  # - b


  # - c

```

**Comment by fisker** [comment]
Maybe add `sequenceItem` will do.

**Review by evilebottnawi — APPROVED** [review]

**Review by lipis — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
@fisker Should we fix https://github.com/prettier/prettier/pull/9143#issuecomment-686955822 on this PR?

**Comment by evilebottnawi** [comment]
@sosukesuzuki I think if it will be easy, let's do it here :smile:

**Comment by fisker** [comment]
I think we should fix it, otherwise it will be `v2.1.0` regression

**Comment by fisker** [comment]
This one seems unreleated, but ugly

**Prettier pr-9143**
[Playground link](https://deploy-preview-9143--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBaABAQwDpV+9AYiwDpIBbcqBGXXDAI3yPQbIkutjqgzGeJh2nGiAA0ICAAcYAS2gBnZKEwAnVRADuABTUIlKTABtNmAJ5KJDVZjABrODADKU27KgBzZDFUBXOBJw5AxwACahYQAymJ6+mB5wAGIQquSYMHKeyCCYvjAQ4iAAFjDkRgDqRbLwCq5gcE761bIAbtVm2WAKliDuCnCqMNo2HmnIAGbG-RIAVgoAHgBCNvaOTpjkcJHucBNTASBz807uHkZwAIq+EPB7RtMgrqr9qtlmG0aFUqruMOWyoRgRWQAA4AAwSb4QfrlGxSbLfOAvFq7CQAR2u8GG0gMOQUqGoYTChVUcAxslJw3iYyQk3uB365Fk3j8DNO5yuN12tP2EhgmAY-0BwKQACY+TZZEZTgBhDg0kBIgCshV8-QAKgKDHSHi1-ABJKARWBOMA-GQAQSNThgZnOd36AF9HUA)
```sh
--parser yaml
```

**Input:**
```yaml
- a

  # a.commnent

- b
  # b.commnent

- c
  # c.commnent
```

**Output:**
```yaml
- a


  # a.commnent
- b

  # b.commnent
- c
  # c.commnent

```

**Comment by sosukesuzuki** [comment]
YAML comment printing seems to have a lot of problems. To be honest, I want to solve the regression only and do patch release as soon as possible... After release, I'll look into those problems. Or, if you already have ideas to solve these, please commit to this branch of mine.

**Review by fisker — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
@fisker Thank you!!

File: tests/yaml/comment/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -336,6 +437,107 @@ printWidth: 80
...
+
+  # before-after.comment
+
+
+- none(2)
+  # none.comment
+=====================================output=====================================
+- before
+
+# before.comment
```
Comment (fisker) [inline tests/yaml/comment/__snapshots__/jsfmt.spec.js.snap:515]: This is strange!!!

**Prettier pr-9143**
[Playground link](https://deploy-preview-9143--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAzCEkB0oAJcDEukAtiQjDgEYCGATtnocRGRTjiADQgQAOMAJbQAzslD06EAO4AFegjEoaAG2k0AnmJ5U6NMAGs4MAMp99gqAHNkMOgFc4POCSpwAJu48AZGtfs0VnAAYhB0JDQwQtbIIDT2MBDcIAAWMCQqAOopgvAi5mBwJoq5ggBuuRqxYCLaIJYicHQwsnpWEcioqo08AFYiAB4AQnqGxiY05N6WcJ3dTiD9AyaWVipwAIr2EPBzKj0g5nSNdLEakyrJfHSWMJmC7jApyAAcAAw81xCNmXp8sdc4CcyrMeABHbbwVr8JRxEQAWigcA8HmSdDgEME6NagQ6SC6+wWjRIglsDiJq3WWx2s3x8x4MBoVHuj2eSAATAy9IIVKsAMKsPEgIEAVmS9kaABUmUoCQcyo4AJJQLywExgG4CACCKpMMA06z2jQAvsagA)
```sh
--parser yaml
```

**Input:**
```yaml
foo:
  # comment
bar:
  # comment


```

**Output:**
```yaml
foo:
  # comment
bar:
  # comment

```

**Prettier pr-9143**
[Playground link](https://deploy-preview-9143--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAzCEkB0oAJcDEukAtiQjDgEYCGATndnocRGRTjiADQgQAOMAJbQAzslD06EAO4AFegnEoaAGxk0AnuN5U6NMAGs4MAMr8DQqAHNkMOgFc4vOCSpwAJh88AZGjYcaazgAMQg6EhoYYRtkEBoHGAgeEAALGBJVAHVUoXhRCzA4UyU8oQA3PM04sFEdECtRODoYOX1rSORUNSbeACtRAA8AIX0jE1Mach8rOC6e5xABwdMra1U4AEUHCHh51V6QCzomujjNKdUU-jorGCyhDxhU5AAOAAZeG4gmrP1+OI3OCncpzXgARx28DaAmU8VEAFooHBPJ4UnQ4JChBi2kFOkhugdFk0SEI7I5iWsNttdnMCQteDAaFQHk8XkgAEyM-RCVRrADCbHxIGBAFYUg4mgAVZnKQmHcpOACSUG8sFMYFuggAgqrTDBNBt9k0AL4moA)
```sh
--parser yaml
```

**Input:**
```yaml
foo:
  # comment
barr:
  # comment


```

**Output:**
```yaml
foo:
# comment
barr:
  # comment

```

Comment (fisker) [reply tests/yaml/comment/__snapshots__/jsfmt.spec.js.snap:515]: Ah, seems attaching logic issue. I'll open an issue in the parser.

**Comment by evilebottnawi** [comment]
@fisker so can we merge this?

**Comment by fisker** [comment]
I think so.

File: tests/yaml/comment/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (fisker) [reply tests/yaml/comment/__snapshots__/jsfmt.spec.js.snap:515]: Issue in `yaml-unist-parser` #275

**Pull request merged**